src/numeric/grids.jl

```diff
@@ -5,19 +5,48 @@ function mlinspace(min, max, n)
     return QE.gridmake(nodes...)
 end
 
-type CartesianGrid <: Grid
-    min::Array{Float64,1}
-    max::Array{Float64,1}
-    n::Array{Int64,1}
+type EmptyGrid <: Grid
+    # this grid does not exist ;-)
 end
 
+nodes(grid::EmptyGrid) = nothing
+n_nodes(grid::EmptyGrid) = 0
+node(grid::EmptyGrid, i::Int64) = nothing # fail if i!=1 ?
 
-type SmolyakGrid <: Grid
-    min::Array{Real,1}
-    max::Array{Real,1}
-    mu::Array{Int,1}
+
+type PointGrid <: Grid
+    point::Vector{Float64}
+end
+nodes(grid::PointGrid) = grid.point[:,:]
```

Comment:
From reading this code, it seems that the other types of grids have that each row is a node across all variables and each column is all nodes for a variable. 

If that is the case, this method is wrong. 

We should define `nodes(grid::PointGrid) = grid.point'` so the output is `(1, n)`